ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp

```diff
@@ -282,6 +292,45 @@ inline auto any_uneven(const Tensor& a, const auto& b, const std::optional<Tenso
     return false;
 }
 
+inline auto is_binary_ng_only(const Tensor& a, const auto& b, BinaryOpType binary_op_type) {
+    if constexpr (requires {
+                      b.dtype();
+                      b.is_sharded();
+                      b.get_logical_shape();
+                  }) {
+        if (a.dtype() == DataType::INT32 or b.dtype() == DataType::INT32 or a.dtype() == DataType::UINT32 or
+            b.dtype() == DataType::UINT32 or a.dtype() == DataType::UINT16 or b.dtype() == DataType::UINT16 or
+            a.dtype() == DataType::UINT8 or b.dtype() == DataType::UINT8) {
+            return true;
+        }
+
+        if (any_row_broadcasted(a, b) and
+            (binary_op_type != BinaryOpType::ADD and binary_op_type != BinaryOpType::SUB and
+             binary_op_type != BinaryOpType::MUL)) {
+            return true;
+        }
+
+        if (a.get_logical_shape().rank() > 4 or b.get_logical_shape().rank() > 4) {
+            return true;
+        }
+
+        if (a.get_logical_shape()[-2] == 1 && b.get_logical_shape()[-2] > 1 && a.get_logical_shape()[-1] > 1 &&
+            b.get_logical_shape()[-1] == 1) {
+            return true;
+        }
+        if (b.get_logical_shape()[-2] == 1 && a.get_logical_shape()[-2] > 1 && b.get_logical_shape()[-1] > 1 &&
+            a.get_logical_shape()[-1] == 1) {
+            return true;
+        }
+
+        if (any_row_broadcasted(a, b) and (is_block_format(a.get_dtype()) or is_block_format(b.get_dtype()))) {
+            // TODO
+            // return true;
+        }
```

Comment:
Would prefer to comment out the entire condition if the early return is to be excluded.